### PR TITLE
Update DecisionReviewCreated to make all calls and link all intake records

### PR DESCRIPTION
**Pull request by buddylove954**
Resolves [Update DecisionReviewCreated to make all calls and link all intake records with rspec updates ](https://jira.devops.va.gov/browse/APPEALS-42637)

# Description
Update the DecisionReviewCreated class to facilitate all necessary calls and establish connections between all intake records.

## Acceptance Criteria

- [x] Uncomment all calls DecisionReviewCreated to child service classes
- [x] ensure proper sequence of creating event records
- [x] ensure proper return types and references are passed to method calls
- [x] Build out Rspec that test each that each method is called with correct parameters
- [x] Rspec should not worry about details of child class implementations and should use doubles
- [x] update the controller to call the DecisionReviewCreated class and pass the JSON body and Header
- [x] update the controller's rpsec to test that the call to DecisionReviewCreated class passes the correct params using doubles

## Testing Plan

1. Go to [43626 for APPEALS-42637](https://jira.devops.va.gov/browse/APPEALS-43626)
2. XRAY for [43626 for APPEALS-42637](https://jira.devops.va.gov/secure/XrayExecuteTest!default.jspa?testExecIssueKey=APPEALS-43628&testIssueKey=APPEALS-43626)

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### CODE REVIEW

File: app/services/events/decision_review_created.rb [hunk 1]
```diff
@@ -28,39 +28,36 @@ def create!(consumer_event_id, reference_id)
         # Use the consumer_event_id to retrieve/create the Event object
         event = find_or_create_event(consumer_event_id)
 
-        # ActiveRecord::Base.transaction do
+        ActiveRecord::Base.transaction do
           # TODO: backfill models as needed, set Event.completed_at when finished
```
Comment (jtsangVA) [inline app/services/events/decision_review_created.rb:32]: remove the "TODO" and update the completed_at column at the end of the method once all the models have been created.

File: app/services/events/decision_review_created.rb [hunk 2]
```diff
@@ -28,39 +28,36 @@ def create!(consumer_event_id, reference_id)
...
 
-          # Note: createdByStation == station_id, createdByUsername == css_id
-          # Events::CreateUserOnEvent.handle_user_creation_on_event(event, css_id, station_id)
-
           # Initialize the Parser object that will be passed around as an argument
-          # parser = Events::DecisionReviewCreated::DecisionReviewCreatedParser.new(headers, payload)
+          parser = Events::DecisionReviewCreated::DecisionReviewCreatedParser.new(headers, payload)
```
Comment (jtsangVA) [inline app/services/events/decision_review_created.rb:35]: These service class calls should have rspec test cases.

File: spec/services/events/decision_review_created_spec.rb [hunk 3]
```diff
@@ -5,6 +5,9 @@
   let!(:reference_id) { "2001" }
   let!(:completed_event) { DecisionReviewCreatedEvent.create!(reference_id: "999", completed_at: Time.zone.now) }
   let!(:event) { instance_double(Event) }
+  let!(:json_payload) { read_json_payload }
+  let!(:headers) { sample_headers }
+
 
   describe "#event_exists_and_is_completed?" do
     subject { described_class.event_exists_and_is_completed?(consumer_event_id) }
```
Comment (jtsangVA) [inline spec/services/events/decision_review_created_spec.rb:13]: flesh out the successful creation section and check that all the expected backfill objects were creat ed

File: spec/services/events/decision_review_created_spec.rb [hunk 4]
```diff
@@ -23,7 +26,7 @@
   end
 
   describe "#create!" do
```
Comment (nkutub) [inline spec/services/events/decision_review_created_spec.rb:28]: Please add checks that each sub service class has been called

File: app/services/events/decision_review_created.rb (hunk 2 above)
Comment (jtsangVA) [reply app/services/events/decision_review_created.rb:35]: I don't see the RequestIssues call, did you pull the latest from feature?

File: app/services/events/decision_review_created.rb [hunk 5]
```diff
@@ -28,39 +28,36 @@ def create!(consumer_event_id, reference_id)
...
+
+          # Note: createdByStation == station_id, createdByUsername == css_id
+          user = Events::CreateUserOnEvent.handle_user_creation_on_event(event, parser.css_id, parser.station_id)
 
           # Create the Veteran. PII Info is stored in the headers
-          # vet = Events::CreateVeteranOnEvent.handle_veteran_creation_on_event(event, parser)
+          vet = Events::CreateVeteranOnEvent.handle_veteran_creation_on_event(event, parser)
```
Comment (nkutub) [inline app/services/events/decision_review_created.rb:41]: can we make this named parameters event: , parser:

File: app/services/events/decision_review_created.rb [hunk 6]
```diff
@@ -28,39 +28,36 @@ def create!(consumer_event_id, reference_id)
...
+
+          # Note Create Claim Review, parsed schema info passed through claim_review and intake
+          decision_review = Events::DecisionReviewCreated::CreateClaimReview.process!(event: event, parser: parser)
 
           # Note: decision_review arg can either be a HLR or SC object. process! will only run if
           # decision_review.legacy_opt_in_approved is true
-          # Events::DecisionReviewCreate::UpdateVacolsOnOptin.process!(decision_review)
-          # event.update!(completed_at: Time.now, error: nil)
+          Events::DecisionReviewCreated::UpdateVacolsOnOptin.process!(decision_review)
```
Comment (nkutub) [inline app/services/events/decision_review_created.rb:48]: make named parameter decision_review:

File: app/services/events/decision_review_created.rb [hunk 7]
```diff
@@ -28,39 +28,36 @@ def create!(consumer_event_id, reference_id)
...
-          # Events::CreateClaimantOnEvent.process(event: event, vbms_claimant: vbms_claimant)
-
-          # Note Create Claim Review, parsed schema info passed through claim_review and intake
-          # Events::DecisionReviewCreated::CreateClaimReview.process!(event: event, parser: parser)
-
+          Events::CreateClaimantOnEvent.process!(event: event, parser: parser,
+                                                 decision_review: decision_review)
           # Note: event, user, and veteran need to be before this call.
-          # Events::DecisionReviewCreated::CreateIntake.process!(event, user, vet)
+          Events::DecisionReviewCreated::CreateIntake.process!(event, user, vet)
```
Comment (nkutub) [inline app/services/events/decision_review_created.rb:55]: make named parameters event:, user: veteran:

File: app/services/events/decision_review_created.rb [hunk 8]
```diff
@@ -28,39 +28,36 @@ def create!(consumer_event_id, reference_id)
...
 
           # Note: end_product_establishment & station_id is coming from the payload
           # claim_review can either be a higher_level_revew or supplemental_claim
-          # Events::DecisionReviewCreated::CreateEpEstablishment.process!(parser, claim_review, user, event)
-
-          # Note: 'epe' arg is the obj created as a result of the CreateEpEstablishment service class
-          # Events::DecisionReviewCreated::CreateRequestIssues.process!(event, parser, epe)
-        # end
+          Events::DecisionReviewCreated::CreateEpEstablishment.process!(parser, decision_review, user, event)
```
Comment (nkutub) [inline app/services/events/decision_review_created.rb:59]: make named parameters parser:, decision_review: , user: , event:

**Review by nkutub — APPROVED** [review]

**Pull request merged**